aten/src/ATen/templates/TensorBody.h

```diff
@@ -86,6 +86,16 @@ inline bool variable_excluded_from_dispatch() {
 // Note that Tensor can also be NULL, i.e. it is not associated with any underlying TensorImpl, and
 // special care must be taken to handle this.
 class TORCH_API Tensor {
+ private:
+  struct unsafe_borrow_t { explicit unsafe_borrow_t() = default; };
```

Comment:
heh, you like this style of named constructor better than the enums we've been using? XD